roles/openshift_sanitize_inventory/tasks/deprecations.yml

```diff
@@ -1,24 +1,27 @@
 ---
 
 - name: Check for usage of deprecated variables
-  set_fact:
-    __deprecation_message: "{{ __deprecation_message | default( __deprecation_header ) }} \n\t{{ item }}"
-  when:
-  - hostvars[inventory_hostname][item] is defined
-  with_items: "{{ __warn_deprecated_vars }}"
+  check_deprecated:
+    facts: "{{ hostvars[inventory_hostname] }}"
+    vars: "{{ __warn_deprecated_vars }}"
+    header: "{{ __deprecation_header }}"
+  register: dep_check
+  run_once: true
 
 - block:
-  - debug: msg="{{__deprecation_message}}"
+  - run_once: true
+    debug: msg="{{ dep_check.message }}"
   - run_once: true
     set_stats:
       data:
         installer_phase_initialize:
-          message: "{{ __deprecation_message }}"
+          message: "{{ dep_check.message }}"
   when:
-  - __deprecation_message | default ('') | length > 0
+  - dep_check.message is defined
```

Comment:
you could set another key in the returned dictionary, like 'deprecations_found' and make it boolean, check it here; or just check if we're 'changed'